test/Microsoft.AspNet.Razor.Runtime.Test/TagHelpers/TagHelperUsageDescriptorFactoryTest.cs

```diff
@@ -0,0 +1,316 @@
+                            remarks: "The remarks for the type.")
+                    },
+                    {
+                        CreateDocumentationTagHelperTestType(location: null, codeBase: defaultCodeBase),
+                        new TagHelperUsageDescriptor(
+                            summary: "The summary for the type.",
+                            remarks: "The remarks for the type.")
+                    },
+                    {
+                        CreateDocumentationTagHelperTestType(defaultLocation, defaultCodeBase),
+                        new TagHelperUsageDescriptor(
+                            summary: "The summary for the type.",
+                            remarks: "The remarks for the type.")
+                    },
+                    { CreateTestType<SingleAttributeTagHelper>(defaultLocation, defaultCodeBase), null },
+                    { CreateDocumentationTagHelperTestType(nonExistentLocation, codeBase: null), null },
+                    { CreateDocumentationTagHelperTestType(location: null, codeBase: nonExistentCodeBase), null },
+                    { CreateTestType<SingleAttributeTagHelper>(invalidLocation, codeBase: null), null },
+                    { CreateDocumentationTagHelperTestType(location: null, codeBase: invalidCodeBase), null },
+                };
+            }
+        }
+
+        public static TheoryData CreateDescriptor_FallbackTypeData
+        {
+            get
+            {
+                // Faking localized dlls only to force the TagHelperUsageDescriptorFactory to search for localized XML.
+                var defaultENGBLocation = DocumentedTestAssemblyLocation.Replace(".dll", "-en-GB.dll");
+                var defaultENLocation = DocumentedTestAssemblyLocation.Replace(".dll", "-en.dll");
+
+                // tagHelperType, expectedUsageDescriptor
+                return new TheoryData<Type, TagHelperUsageDescriptor>
+                {
+                    {
+                        CreateDocumentationTagHelperTestType(defaultENGBLocation, codeBase: null),
+                        new TagHelperUsageDescriptor(
+                            summary: "en-GB: The summary for the type.",
+                            remarks: "en-GB: The remarks for the type.")
+                    },
+                    {
+                        CreateDocumentationTagHelperTestType(defaultENLocation, codeBase: null),
+                        new TagHelperUsageDescriptor(
+                            summary: "en: The summary for the type.",
+                            remarks: "en: The remarks for the type.")
+                    }
+                };
+            }
+        }
+
+        [Theory]
+        [ReplaceCulture]
+        [MemberData(nameof(CreateDescriptor_TypeData))]
+        [MemberData(nameof(CreateDescriptor_FallbackTypeData))]
+        public void CreateDescriptor_Type(Type tagHelperType, TagHelperUsageDescriptor expectedUsageDescriptor)
```

Comment:
Change all tests in the two new classes to have names that describe expectations. This one might be `CreateDescriptor_WithType_ReturnsExpectedDescriptors()`.